financialaid/views.py

```diff
@@ -74,30 +75,43 @@ class FinancialAidSkipView(UpdateAPIView):
     aid object exists, and then either creates or updates a financial aid object to reflect
     the user skipping financial aid.
     """
-    serializer_class = FinancialAidSkipSerializer
     authentication_classes = (
         SessionAuthentication,
         TokenAuthentication,
     )
     permission_classes = (IsAuthenticated, )
 
-    def get_object(self):
+    def update(self, request, *args, **kwargs):
         """
         Overrides get_object in case financialaid object does not exist, as the learner may skip
         financial aid either after starting the process or in lieu of applying
         """
+        user = request.user
         program = get_object_or_404(Program, id=self.kwargs["program_id"])
         if not program.financial_aid_availability:
             raise ValidationError("Financial aid not available for this program.")
-        if not ProgramEnrollment.objects.filter(program=program.id, user=self.request.user).exists():
+        if not ProgramEnrollment.objects.filter(program=program.id, user=user).exists():
             raise ValidationError("User not in program.")
-        tier_program = get_no_discount_tier_program(program.id)
-        financialaid, _ = FinancialAid.objects.get_or_create(
-            user=self.request.user,
+
+        financialaid = FinancialAid.objects.filter(
+            user=user,
             tier_program__program=program,
-            defaults={"tier_program": tier_program}
-        )
-        return financialaid
+        ).exclude(status=FinancialAidStatus.RESET).first()
+        if financialaid is None:
+            financialaid = FinancialAid(
+                user=user,
+                country_of_income=user.profile.country,
+                date_exchange_rate=datetime.datetime.now(),
+                country_of_residence=user.profile.country,
+            )
+
+        if financialaid.status in FinancialAidStatus.TERMINAL_STATUSES:
+            raise ValidationError("Financial aid application cannot be skipped once it's been approved or skipped.")
+
+        financialaid.tier_program = get_no_discount_tier_program(program.id)
+        financialaid.status = FinancialAidStatus.SKIPPED
+        financialaid.save_and_log(user)
+        return Response(status=HTTP_200_OK)
```

Comment:
I'd prefer not to since this isn't actually a `FinancialAid` endpoint but instead something specific to skip requests. If we're going to make it more RESTful we should just combine all the APIs into one first